### PR TITLE
IFTAQCAB-40 Add new speciality

### PR DESCRIPTION
**Pull request by promahos**

### CODE REVIEW

File: src/main/java/com/softserveinc/ita/pageobjects/admin/SpecialitiesPage.java [hunk 1]
```diff
@@ -1,4 +1,42 @@
...
+import static com.codeborne.selenide.Selenide.*;
+
 public class SpecialitiesPage extends MainMenu {
+
+    @Step("Speciality page: Added new speciality")
+    public SpecialitiesPage addNewSpeciality(String code, String name) {
+        $x("//button[contains(@class,'addButton')]").click();
+        $x("//input[@formcontrolname='speciality_code']").setValue(code);
+        $x("//input[@formcontrolname='speciality_name']").setValue(name);
+        $$x("//button/span")
```
Comment (Nikodych) [inline src/main/java/com/softserveinc/ita/pageobjects/admin/SpecialitiesPage.java:16]: just use a direct locator //button/span[contains(text(),'Підтвердити')]

**Review by Nikodych — CHANGES_REQUESTED** [review]

File: src/main/java/com/softserveinc/ita/pageobjects/util/RandomUtil.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+import static java.lang.Math.pow;
+
+@UtilityClass
+public class RandomUtil {
+    public int getRandomNumber(int length) {
+        int bound; // bound is max possible value
+
+        if (length > 9) {
+            bound = Integer.MAX_VALUE;
```
Comment (Nikodych) [inline src/main/java/com/softserveinc/ita/pageobjects/util/RandomUtil.java:15]: stat import of MAX_VALUE

File: src/main/java/com/softserveinc/ita/pageobjects/util/RandomUtil.java [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+import lombok.experimental.UtilityClass;
+
+import java.util.Random;
+
+import static java.lang.Math.pow;
+
+@UtilityClass
+public class RandomUtil {
+    public int getRandomNumber(int length) {
```
Comment (Nikodych) [inline src/main/java/com/softserveinc/ita/pageobjects/util/RandomUtil.java:11]: make it static

File: src/test/java/com/softserveinc/ita/SpecialitiesTest.java [hunk 4]
```diff
@@ -38,9 +36,10 @@ public void verifySpecialitiesPageOpening() {
     }
 
     @Test
-    public void verifyNewSpecialityIsAdded() {
+    @Description("Test to verify new speciality is added")
+    public void verifyAddingNewSpeciality() {
 
-        var randCode = new Random().nextInt(99999);
+        var randCode = RandomUtil.getRandomNumber(5);
```
Comment (Nikodych) [inline src/test/java/com/softserveinc/ita/SpecialitiesTest.java:42]: static import of "getRandomNumber(5)"

**Review by Nikodych — CHANGES_REQUESTED** [review]

File: src/main/java/com/softserveinc/ita/pageobjects/util/RandomUtil.java (hunk 3 above)
Comment (promahos) [reply src/main/java/com/softserveinc/ita/pageobjects/util/RandomUtil.java:11]: All members of a utility class are automatically marked as static with lombok @UtilityClass annotation

**Review by Nikodych — APPROVED** [review]

**Pull request merged**